cpython-unix/build-main.py

```diff
@@ -54,10 +55,17 @@ def main():
         help="Target host triple to build for",
     )
 
-    optimizations = {"debug", "noopt", "pgo", "lto", "pgo+lto"}
+    # Construct possible options, we use a set here for canonical ordering
+    options = set()
+    options.update({"debug", "noopt", "pgo", "lto", "pgo+lto"})
+    options.update({f"freethreaded+{option}" for option in options})
+    link_modes = {"static", "shared"}
+    options.update(
+        {f"{option}+{link_mode}" for link_mode in link_modes for option in options}
+    )
```

Comment:
I don't like this, especially since it's repeated. I will refactor this further separately.